src/R/Components/Test/PackageManager/PackageManagerIntegrationTest.cs

```diff
@@ -179,6 +179,145 @@ public class PackageManagerIntegrationTest : IDisposable, IAsyncLifetime {
+                installed.Should().ContainSingle(pkg => pkg.Package == TestPackages.RtvsLib1Description.Package && pkg.LibPath == _libPath.ToRPath());
+            }
+        }
+
+        [Test]
+        [Category.PackageManager]
+        public async Task LoadAndUnloadPackage() {
+            var interactiveWindowComponentContainerFactory = _exportProvider.GetExportedValue<IInteractiveWindowComponentContainerFactory>();
+            using (await UIThreadHelper.Instance.Invoke(() => _workflow.GetOrCreateVisualComponent(interactiveWindowComponentContainerFactory))) {
+                _workflow.ActiveWindow.Should().NotBeNull();
+                _workflow.RSession.IsHostRunning.Should().BeTrue();
+
+                using (var eval = await _workflow.RSession.BeginEvaluationAsync()) {
+                    await SetLocalRepoAsync(eval, _repo1Path);
+                    await SetLocalLibAsync(eval, _libPath);
+                    await InstallPackageAsync(eval, TestPackages.RtvsLib1Description.Package, _libPath);
+                }
+
+                await EvaluateCode("func1()", expectedError: "Error: could not find function \"func1\"");
+
+                _workflow.Packages.LoadPackage(TestPackages.RtvsLib1Description.Package, null);
+                WaitForReplDoneExecuting();
+
+                await EvaluateCode("func1()", expectedResult: "func1");
+
+                var loaded = await _workflow.Packages.GetLoadedPackagesAsync();
+                loaded.Should().Contain("rtvslib1");
+
+                _workflow.Packages.UnloadPackage(TestPackages.RtvsLib1Description.Package);
+                WaitForReplDoneExecuting();
+
+                await EvaluateCode("func1()", expectedError: "Error: could not find function \"func1\"");
+
+                loaded = await _workflow.Packages.GetLoadedPackagesAsync();
+                loaded.Should().NotContain("rtvslib1");
+            }
+        }
+
+        [Test]
+        [Category.PackageManager]
+        public async Task GetLoadedPackages() {
+            var interactiveWindowComponentContainerFactory = _exportProvider.GetExportedValue<IInteractiveWindowComponentContainerFactory>();
+            using (await UIThreadHelper.Instance.Invoke(() => _workflow.GetOrCreateVisualComponent(interactiveWindowComponentContainerFactory))) {
+                _workflow.ActiveWindow.Should().NotBeNull();
+                _workflow.RSession.IsHostRunning.Should().BeTrue();
+
+                using (var eval = await _workflow.RSession.BeginEvaluationAsync()) {
+                    await SetLocalRepoAsync(eval, _repo1Path);
+                    await SetLocalLibAsync(eval, _libPath);
+                    await InstallPackageAsync(eval, TestPackages.RtvsLib1Description.Package, _libPath);
+                }
+
+                string[] results = await _workflow.Packages.GetLoadedPackagesAsync();
+                results.Should().NotContain("rtvslib1");
```

Comment:
There are overloads of `Contain` and `NotContain` that take an enumerable, and check for all elements in it (order doesn't matter, so it's not a subsequence match, just a subset check).